x/dclauth/handler_test.go

```diff
@@ -1373,6 +1400,178 @@ func TestHandler_CreateVendorAccountWithDifferentRole_ThreeApprovalsAreNeeded(t
+
+	// trustee1 propose account
+	_, address, pubKey, err := proposeAddAccount(setup, trustee1, types.AccountRoles{types.NodeAdmin, types.Vendor})
+	require.NoError(t, err)
+
+	// approve account by account Trustee2
+	approveAddAccount := types.NewMsgApproveAddAccount(trustee2, address, testconstants.Info2)
+	_, err = setup.Handler(setup.Ctx, approveAddAccount)
+	require.NoError(t, err)
+
+	// approve account by account Trustee3
+	approveAddAccount = types.NewMsgApproveAddAccount(trustee3, address, testconstants.Info2)
+	_, err = setup.Handler(setup.Ctx, approveAddAccount)
+	require.NoError(t, err)
+
+	// account should be in the entity <Proposed Account>, because we haven't enough approvals
+	proposedAccount, found := setup.Keeper.GetPendingAccount(setup.Ctx, address)
+	require.True(t, found)
+
+	// check proposed account
+	require.Equal(t, address.String(), proposedAccount.Address)
+	require.Equal(t, pubKey, proposedAccount.GetPubKey())
+
+	// approve account by account Trustee4
+	approveAddAccount = types.NewMsgApproveAddAccount(trustee4, address, testconstants.Info2)
+	_, err = setup.Handler(setup.Ctx, approveAddAccount)
+	require.NoError(t, err)
+
+	// account should be in the entity <Account>, because we have enough approvals
+	approvedAccount, found := setup.Keeper.GetAccountO(setup.Ctx, address)
+	require.True(t, found)
+
+	// check account
+	require.Equal(t, address.String(), approvedAccount.Address)
+	require.Equal(t, pubKey, approvedAccount.GetPubKey())
+}
+
+func TestHandler_ApproveVendorAccount_TwoApprovalsAreNeeded_FourTrustees(t *testing.T) {
+	setup := Setup(t)
+
+	// we have 4 trustees: 1 approval comes from propose => we need 1 more approval
+
+	trustee1 := storeTrustee(setup)
+	trustee2 := storeTrustee(setup)
+	_ = storeTrustee(setup)
+	_ = storeTrustee(setup)
+
+	// trustee1 propose account
+	_, address, pubKey, err := proposeAddAccount(setup, trustee1, types.AccountRoles{types.Vendor})
+	require.NoError(t, err)
+
+	// account should be in the entity <Proposed Account>, because we haven't enough approvals
+	proposedAccount, found := setup.Keeper.GetPendingAccount(setup.Ctx, address)
+	require.True(t, found)
+
+	// check proposed account
+	require.Equal(t, address.String(), proposedAccount.Address)
+	require.Equal(t, pubKey, proposedAccount.GetPubKey())
+
+	// approve account by account Trustee2
+	approveAddAccount := types.NewMsgApproveAddAccount(trustee2, address, testconstants.Info2)
+	_, err = setup.Handler(setup.Ctx, approveAddAccount)
+	require.NoError(t, err)
+
+	// account should be in the entity <Account>, because we have enough approvals
+	approvedAccount, found := setup.Keeper.GetAccountO(setup.Ctx, address)
+	require.True(t, found)
+
+	// check account
+	require.Equal(t, address.String(), approvedAccount.Address)
+	require.Equal(t, pubKey, approvedAccount.GetPubKey())
+}
+
+func TestHandler_RejectVendorAccount_TwoRejectsAreNeeded_FourTrustees(t *testing.T) {
```

Comment:
Can we please make a test for rejecting a vendor account when we have 5 trustees?